packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -54,20 +118,48 @@ class CupertinoButton extends StatefulWidget {
   const CupertinoButton({
     super.key,
     required this.child,
+    this.sizeStyle = CupertinoButtonSize.large,
     this.padding,
     this.color,
     this.disabledColor = CupertinoColors.quaternarySystemFill,
-    this.minSize = kMinInteractiveDimensionCupertino,
+    this.minSize,
     this.pressedOpacity = 0.4,
-    this.borderRadius = const BorderRadius.all(Radius.circular(8.0)),
+    this.borderRadius,
     this.alignment = Alignment.center,
     this.focusColor,
     this.focusNode,
     this.onFocusChange,
     this.autofocus = false,
     required this.onPressed,
   }) : assert(pressedOpacity == null || (pressedOpacity >= 0.0 && pressedOpacity <= 1.0)),
-       _filled = false;
+       _style = _CupertinoButtonStyle.plain;
+
+  /// Creates an iOS-style button with a tinted background.
+  ///
+  /// The background color is derived from the [CupertinoTheme]'s `primaryColor` + transparency.
+  /// The foreground color is the [CupertinoTheme]'s `primaryColor`.
+  ///
+  /// To specify a custom background color, use the [color] argument of the
+  /// default constructor.
+  ///
+  /// To match the iOS "grey" button style, set [color] to [CupertinoColors.systemGrey].
+  const CupertinoButton.tinted({
+    super.key,
+    required this.child,
+    this.sizeStyle = CupertinoButtonSize.large,
+    this.padding,
+    this.color,
+    this.disabledColor = CupertinoColors.quaternarySystemFill,
```

Comment:
Does this default disabled color also match native iOS in dark mode? If not, consider using CupertinoDynamicColor.resolve to make it resolve differently in light vs. dark modes.